### PR TITLE
[DR-83]Button animations

### PR DESCRIPTION
**Pull request by mgolawski**
[DR-83](https://digitalrig.atlassian.net/browse/DR-83) Buttons animations

# Description 
ScreenRecording2019-07-03at1

## Checklist:

- [ ] Icon Bounce
- [ ] Button onClick bounce effect

### CODE REVIEW

File: src/components/ActionButtons/ButtonBase.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { func, oneOfType, object, node } from 'prop-types';
+import { useSpring, interpolate } from 'react-spring';
+
+import Button from '../Button';
+
+function ButtonBase({ Icon, onClick = () => {}, children, ...props }) {
```
Comment (blurbyte) [inline src/components/ActionButtons/ButtonBase.js]: I would change **I**con to **i**con and afterwards make an alias `icon: Icon`. Capital name of prop looks really weird, even if we consider that it expects component.

File: src/components/ActionButtons/ButtonBase.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const [shouldPlayAnimation, toggleIconAnmiation] = useState(false);
+  const handleClick = e => {
```
Comment (blurbyte) [inline src/components/ActionButtons/ButtonBase.js]: In most cases we use full word **event**

File: src/components/ActionButtons/ButtonBase.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    onClick(e);
+  };
+
+  const { x } = useSpring({
+    x: shouldPlayAnimation ? 1 : 0,
+    onRest: () => toggleIconAnmiation(false)
+  });
+
+  const iconShakingAnimation = {
+    transform: !shouldPlayAnimation
```
Comment (blurbyte) [inline src/components/ActionButtons/ButtonBase.js]: Maybe `transform: !shouldPlayAnimation && interpolate(...)` will work?

File: src/components/Button/Button.js [hunk 4]
```diff
@@ -1,36 +1,42 @@
...
+    ...props,
+    onClick,
+    ...(to ? { as: Link, to } : { onClick })
+  };
 
-  &:not(:disabled) {
-    cursor: pointer;
-  }
-`;
+  const [isClicked, setClicked] = useState(false);
```
Comment (blurbyte) [inline src/components/Button/Button.js]: In most cases just prefix value with set therefore **setIsClicked**.

File: src/components/ActionButtons/ButtonBase.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { func, oneOfType, object, node } from 'prop-types';
```
Comment (blurbyte) [inline src/components/ActionButtons/ButtonBase.js]: Besides centralised prop types I think we don't destructure PropTypes.

File: src/components/Button/Button.js [hunk 6]
```diff
@@ -1,36 +1,42 @@
...
+      tension: 500,
+      friction: 8
+    }
+  });
+
+  return (
+    <StyledButton
+      {...mappedProps}
+      style={animationStyles}
+      onMouseDown={() => setClicked(true)}
```
Comment (blurbyte) [inline src/components/Button/Button.js]: Maybe use similar approach as in one of previous animations with **onRest** callback?

Comment (mgolawski) [reply src/components/Button/Button.js]: Using this approach holding click on button won't work. Trigger of reverse animation is strictly connected with MouseUp event which affects positively the animation timing.

**Pull request merged**